src/Microsoft.OData.Core/UriParser/SemanticAst/ExpandedCountSelectItem.cs

```diff
@@ -0,0 +1,53 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ExpandedCountSelectItem.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.UriParser
+{
+    using Aggregation;
+    using Microsoft.OData.Edm;
+
+    /// <summary>
+    /// This represents one level of expansion for a particular expansion tree.
+    /// </summary>
+    public sealed class ExpandedCountSelectItem : ExpandedReferenceSelectItem
+    {
+        /// <summary>
+        /// Create an Expand item using a nav prop, its entity set and a SelectExpandClause
+        /// </summary>
+        /// <param name="pathToNavigationProperty">the path to the navigation property for this expand item, including any type segments</param>
+        /// <param name="navigationSource">the navigation source for this ExpandItem</param>
+        /// <param name="filterOption">A filter clause for this expand (can be null)</param>
+        /// <param name="searchOption">A search clause for this expand (can be null)</param>
+        /// <exception cref="System.ArgumentNullException">Throws if input pathToNavigationProperty is null.</exception>
+        public ExpandedCountSelectItem(ODataExpandPath pathToNavigationProperty, IEdmNavigationSource navigationSource, FilterClause filterOption, SearchClause searchOption)

```

Comment:
Only $filter and $search are allowed within nav/$count?

Reply:
Yes: From the protocol http://docs.oasis-open.org/odata/odata/v4.01/cs01/part1-protocol/odata-v4.01-cs01-part1-protocol.html#sec_RequestingtheNumberofItemsinaCollect

_On success, the response body MUST contain the exact count of items matching the request after applying any $filter or $search system query options, formatted as a simple primitive integer value with media type text/plain. Clients SHOULD NOT combine the system query options  $top, $skip, $orderby, $expand, and $format with the path suffix /$count. The result of such a request is undefined._